lahja/endpoint.py

```diff
@@ -33,62 +38,102 @@
 )
 
 
+class ConnectionConfig(NamedTuple):
```

Comment:
After looking through the tests, it feels like requiring consumers to pass `ConnectionConfig`s rather than names imposes a large cost on them for what is essentially an implementation detail (which Path to connect to). Maybe you expect to add more details to `ConnectionConfig`s in the future? If `Endpoints` will usually be referred to by names, what do you think of passing those names directly into methods like `connect_no_wait` and having them derive the appropriate Path themselves?

Reply:
>what is essentially an implementation detail (which Path to connect to)

It is for many cases which is why `from_name` exists to make it smoother for the happy path where you don't care about the ipc path. However, e.g. in Trinity we want all IPC paths to go into a certain directory.

Reply:
Gotcha, that makes sense.

Is there another way to make this interface simpler? Maybe something like:

```python
# the caller keeps track of this
ctx = lahja.ConnectionContext('/tmp/ipc_dir') 

# so that later it can call this:
endpoint = ctx.Endpoint('peerpool')
```

Callers would lose control over the exact layout of the ipcs but would still be able to specify a directory to use and Endpoints would be very easy to create and reference. This would also allow for something like a `MemoryContext` for running quick tests without creating ipc files.

I don't know, that sounds like it has its own drawbacks. Definitely feel free to ignore me but it feels heavyweight to specify the name and exact path every time you want to reference an Endpoint when one could be used to derive the other.

Reply:
We could probably keep track of a directory as you propose but it would still give up a bit flexibility. Different endpoints may reside in entirely different directories. I'm not sure if we really want to give up on that flexibility.

>it feels heavyweight to specify the name and exact path every time you want to reference an Endpoint

Which really you don't need if you don't care about the path. Then you just use `ConnectionConfig.from_name(`foo`)` and it will derive it on it's own.

But I guess we could come up with a scheme that allows both fine-grained case by case as well as directory based path resolution.

But considering that most applications will only need these connect calls in very few places it doesn't catch me that strongly.